### PR TITLE
lab-12-1-rnn

### PR DESCRIPTION
**Pull request by jennykang**

### CODE REVIEW

File: lab-12-1-rnn.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+input_dim = 5
+
+x_data = np.array([[h, e, l, l, o]], dtype=np.float32)
+print(x_data)
+
+cell = rnn.BasicLSTMCell(num_units=hidden_size, state_is_tuple=True)
+initial_state = cell.zero_state(batch_size, tf.float32)
+outputs, _states = tf.nn.dynamic_rnn(cell, x_data, initial_state=initial_state, dtype=tf.float32)
+
+x_data = x_data.reshape(-1, hidden_size)
```
Comment (hunkim) [inline lab-12-1-rnn.py]: remove softmax here. Just directly use output from LSTM.

File: lab-12-1-rnn.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+print(x_data)
+softmax_w = np.arange(20, dtype=np.float32).reshape(hidden_size, num_classes)
+print(x_data.shape)
+print(softmax_w.shape)
+outputs = np.matmul(x_data, softmax_w)
+outputs = outputs.reshape(-1, sequence_length, num_classes)
+
+
+prediction = tf.constant([[h, e, l, l, o]], dtype=tf.float32)
+y_data = tf.constant([[1, 1, 1, 1, 1]])
```
Comment (hunkim) [inline lab-12-1-rnn.py]: y_data is the label: e l l o, not 1.1.1.1

File: lab-12-1-rnn.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+weights = tf.constant([[1, 1, 1, 1, 1]], dtype=tf.float32)
```
Comment (hunkim) [inline lab-12-1-rnn.py]: Do you understand what is the weight?

File: lab-12-1-rnn.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+pp = pprint.PrettyPrinter(indent=4)
+
+sess = tf.InteractiveSession()
+
+h = [1,0,0,0]
+e = [0,1,0,0]
+l = [0,0,1,0]
+o = [0,0,0,1]
+
+hidden_size = 5
```
Comment (hunkim) [inline lab-12-1-rnn.py]: Why hidden size is 5?

File: lab-12-1-rnn.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+input_dim = 5
+
+x_data = np.array([[h, e, l, l, o]], dtype=np.float32)
+print(x_data)
+
+cell = rnn.BasicLSTMCell(num_units=hidden_size, state_is_tuple=True)
+initial_state = cell.zero_state(batch_size, tf.float32)
+outputs, _states = tf.nn.dynamic_rnn(cell, x_data, initial_state=initial_state, dtype=tf.float32)
+
+prediction = tf.constant([[h, e, l, l, o]], dtype=tf.float32)
```
Comment (hunkim) [inline lab-12-1-rnn.py]: prediction should come from lstm output.
